kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java

```diff
@@ -700,7 +700,9 @@ SSLConnectionSocketFactory getTwoWaySSLSocketFactory() throws SQLException {
   }
 
   private void openSession() throws SQLException {
-    TOpenSessionReq openReq = new TOpenSessionReq();
+    // Server responses are the same if TProtocolVersion >= HIVE_CLI_SERVICE_PROTOCOL_V6
```

Comment:
also cc Hive experts @wangyum

Reply:
Kyuubi JDBC uses `HIVE_CLI_SERVICE_PROTOCOL_V10`,
`TOpenSessionReq.TOpenSessionReqStandardScheme#read` returns null because there is no corresponding protocol version.
So there is `Required field 'client_protocol' is unset! ` in the PR description.

https://github.com/apache/hive/blob/b02cef4fe943b9aba597dcdfd3b8f3d3a5efca3e/service-rpc/src/gen/thrift/gen-javabean/org/apache/hive/service/rpc/thrift/TOpenSessionReq.java#L580

If Kyuubi JDBC uses a lower version, HS2 will compare the currently known version, find the minimum version number, and return it to the Client.

https://github.com/apache/hive/blob/b02cef4fe943b9aba597dcdfd3b8f3d3a5efca3e/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java#L556-L557

Reply:
@cxzl25 Except `TOpenSessionReq`, is there any difference in responses of other requests  between different protocol versions ?

Reply:
Kyuubi currently determines whether to return TIMEDOUT_STATE based on the client's protocol version.
https://github.com/apache/kyuubi/issues/2112

v7
0.13.0
[HIVE-6647](https://issues.apache.org/jira/browse/HIVE-6647) : Bump the thrift api version to V7 for HiveServer2

v8
1.2.0
[HIVE-10037](https://issues.apache.org/jira/browse/HIVE-10037): JDBC support for interval expressions

v9
2.2.0
[HIVE-12427](https://issues.apache.org/jira/browse/HIVE-12427): ResultSet performance
[HIVE-14191](https://issues.apache.org/jira/browse/HIVE-14191): bump a new api version for ThriftJDBCBinarySerde changes

org.apache.hive.service.cli.session.HiveSessionImpl#updateIsUsingThriftJDBCBinarySerDe
```java
  private boolean updateIsUsingThriftJDBCBinarySerDe() {
    return 8 <= getProtocolVersion().getValue() &&
      sessionConf.getBoolVar(ConfVars.HIVE_SERVER2_THRIFT_RESULTSET_SERIALIZE_IN_TASKS);
  }
```

v10
2.2.0
[HIVE-15473](https://issues.apache.org/jira/browse/HIVE-15473): Progress Bar on Beeline client
[HIVE-15906](https://issues.apache.org/jira/browse/HIVE-15906) : thrift code regeneration to include new protocol version

v11
3.0.0
[HIVE-14412](https://issues.apache.org/jira/browse/HIVE-14412): Add timestamp with time zone

Reply:
I suppose such change is only affected by HIVE-14191?